src/components/src/main/java/org/apache/jmeter/config/CSVDataSet.java

```diff
@@ -222,20 +222,21 @@ private void initVars(FileServer server, final JMeterContext context, String del
     private void setAlias(final JMeterContext context, String alias) {
         String mode = getShareMode();
         int modeInt = CSVDataSetBeanInfo.getShareModeAsInt(mode);
-        switch(modeInt){
-            case CSVDataSetBeanInfo.SHARE_ALL:
-                this.alias = alias;
-                break;
-            case CSVDataSetBeanInfo.SHARE_GROUP:
-                this.alias = alias + "@" + System.identityHashCode(context.getThreadGroup());
-                break;
-            case CSVDataSetBeanInfo.SHARE_THREAD:
-                this.alias = alias + "@" + System.identityHashCode(context.getThread());
-                break;
-            default:
-                this.alias = alias + "@" + mode; // user-specified key
-                break;
-        }
+        this.alias = switch(modeInt){
+            case CSVDataSetBeanInfo.SHARE_ALL -> alias;
+            case CSVDataSetBeanInfo.SHARE_GROUP -> {
+                // Complex logic for thread group sharing
+                yield alias + "@" + System.identityHashCode(context.getThreadGroup());
```

Comment:
just wondering, is `yield` needed here?

Reply:
Thanks for comment, change in last PR version